File: CivitasJava/civitas/CasillaSorpresa.java
package civitas;

import java.util.ArrayList;

public class CasillaSorpresa extends Casilla {
    MazoSorpresas mazo;
    Sorpresa sorpresa;
    
    CasillaSorpresa(MazoSorpresas m, String s){
        super(s);
        mazo = m;
    }
    
    @Override
    void recibeJugador (int iactual, ArrayList<Jugador> todos) throws Exception{
        if (jugadorCorrecto(iactual, todos)){
            informe(iactual, todos);
            mazo.siguiente().aplicarAJugador(iactual, todos);
	}
    }
}

File: CivitasRuby/casilla_sorpresa.rb
require_relative 'casilla'

module Civitas
class CasillaSorpresa < Casilla
  def initialize(mazo, nombre)
    super(nombre)
    @mazo = mazo
  end
  
  def recibe_jugador(actual, todos)
    if jugador_correcto(actual, todos)
      sorpresa = @mazo.siguiente
      informe(actual, todos)
      sorpresa.aplicar_a_jugador(actual, todos)
    end
  end
end
end
